packages/tools/addon/components/cs-version-control.js

```diff
@@ -158,6 +159,15 @@ export default Component.extend({
     yield model.save();
   }).keepLatest(),
 
+  cancel: task(function* () {
+    if (this.anythingDirty) {
+      let model = this.get('model');
+      let upstreamModel = yield this.get('store').findRecord(model.type, model.id);
```

Comment:
I think this could be synchronous and use `peekRecord` as only models that are already in the store could even be dirty at all.

Reply:
When I use `peekRecord`, the cancel button doesn't undo the changes. Any idea?

Reply:
Hm, that seems strange. I just realized you should also be able to simply do `model.rollbackAttributes` though…

Reply:
That's what I initially tried as well, but simply doing that didn't revert the changes. Even though the changes aren't actually saved, visually the changes remain until page is refreshed. Only when I fetch the record, it's reset to the formerly saved version without needing to refresh the page.

Reply:
Also realized changes in the dropdown menu selection (ie. selecting a different Artist using the dropdown) doesn't mark the field as dirty. Will take a look at that. Multi select dropdown is fine (Genre selection).